### ISSUE TITLE
Tokio panic with Unordered

### ISSUE THREAD

**Issue by Tockra** [stevepryde/thirtyfour#48]
Hi Guys,

this lines causes an panic:
```rust
 let mut filtered_needed_elements = FuturesUnordered::new();
    for element in needed_elements.into_iter() {
        filtered_needed_elements.push(driver.find_element(By::XPath(element)));
    }
    while let Some(filtered_needed_element) = filtered_needed_elements.next().await {
        if filtered_needed_element.is_ok() {
            return Ok(false);
        }
    }
```
This panic occurs somewhen after executing this lines.

Here the stacktrace:
```bash
thread 'tokio-runtime-worker' panicked at 'Failed to send response: Err(NoSuchElement(WebDriverErrorInfo { status: 404, error: "", value: WebDriverErrorValue { message: "Unable to locate element: //*[@id=\"buy-now-button\"]", error: Some("no such element"), stacktrace: Some("WebDriverError@chrome://marionette/content/error.js:181:5\nNoSuchElementError@chrome://marionette/content/error.js:393:5\nelement.find/</<@chrome://marionette/content/element.js:306:16\n"), data: None } }))', C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\thirtyfour-0.23.0\src\session.rs:47:30
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b\/library\std\src\panicking.rs:493
   1: core::panicking::panic_fmt
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b\/library\core\src\panicking.rs:92
   2: core::option::expect_none_failed
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b\/library\core\src\option.rs:1268
   3: core::result::Result<tuple<>, core::result::Result<serde_json::value::Value, thirtyfour::error::WebDriverError>>::expect<tuple<>,core::result::Result<serde_json::value::Value, thirtyfour::error::WebDriverError>>        
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\result.rs:933
   4: thirtyfour::session::session_runner::{{closure}}
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\thirtyfour-0.23.0\src\session.rs:47
   5: core::future::from_generator::{{impl}}::poll<generator-0>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\future\mod.rs:80
   6: tokio::runtime::task::core::{{impl}}::poll::{{closure}}<core::future::from_generator::GenFuture<generator-0>,alloc::sync::Arc<tokio::runtime::thread_pool::worker::Worker>>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\core.rs:173
   7: tokio::loom::std::unsafe_cell::UnsafeCell<tokio::runtime::task::core::Stage<core::future::from_generator::GenFuture<generator-0>>>::with_mut<tokio::runtime::task::core::Stage<core::future::from_generator::GenFuture<generator-0>>,core::task::poll::Poll<tup
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\loom\std\unsafe_cell.rs:14
   8: tokio::runtime::task::core::Core<core::future::from_generator::GenFuture<generator-0>, alloc::sync::Arc<tokio::runtime::thread_pool::worker::Worker>>::poll<core::future::from_generator::GenFuture<generator-0>,alloc::sync::Arc<tokio::runtime::thread_pool::
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\core.rs:158
   9: tokio::runtime::task::harness::{{impl}}::poll::{{closure}}<core::future::from_generator::GenFuture<generator-0>,alloc::sync::Arc<tokio::runtime::thread_pool::worker::Worker>>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\harness.rs:107
  10: core::ops::function::FnOnce::call_once<closure-0,tuple<>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
  11: std::panic::{{impl}}::call_once<core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>,closure-0>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:322
  12: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:379
  13: thirtyfour::webdrivercommands::start_session::{{closure}}::_::{{impl}}::deserialize::{{impl}}::expecting
  14: std::panicking::try<core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:343
  15: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:396
  16: tokio::runtime::task::harness::Harness<core::future::from_generator::GenFuture<generator-0>, alloc::sync::Arc<tokio::runtime::thread_pool::worker::Worker>>::poll<core::future::from_generator::GenFuture<generator-0>,alloc::sync::Arc<tokio::runtime::thread_
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\harness.rs:89
  17: tokio::runtime::task::raw::poll<core::future::from_generator::GenFuture<generator-0>,alloc::sync::Arc<tokio::runtime::thread_pool::worker::Worker>>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\raw.rs:104
  18: tokio::runtime::task::raw::RawTask::poll
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\raw.rs:66
  19: tokio::runtime::task::Notified<alloc::sync::Arc<tokio::runtime::thread_pool::worker::Worker>>::run<alloc::sync::Arc<tokio::runtime::thread_pool::worker::Worker>>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\mod.rs:171
  20: tokio::runtime::thread_pool::worker::{{impl}}::run_task::{{closure}}
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\thread_pool\worker.rs:370
  21: tokio::coop::with_budget::{{closure}}<core::result::Result<alloc::boxed::Box<tokio::runtime::thread_pool::worker::Core, alloc::alloc::Global>, tuple<>>,closure-0>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\coop.rs:121
  22: std::thread::local::LocalKey<core::cell::Cell<tokio::coop::Budget>>::try_with<core::cell::Cell<tokio::coop::Budget>,closure-0,core::result::Result<alloc::boxed::Box<tokio::runtime::thread_pool::worker::Core, alloc::alloc::Global>, tuple<>>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\thread\local.rs:272
  23: std::thread::local::LocalKey<core::cell::Cell<tokio::coop::Budget>>::with<core::cell::Cell<tokio::coop::Budget>,closure-0,core::result::Result<alloc::boxed::Box<tokio::runtime::thread_pool::worker::Core, alloc::alloc::Global>, tuple<>>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\thread\local.rs:248
  24: tokio::coop::with_budget
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\coop.rs:114
  25: tokio::coop::budget
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\coop.rs:98
  26: tokio::runtime::thread_pool::worker::Context::run_task
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\thread_pool\worker.rs:348
  27: tokio::runtime::thread_pool::worker::Context::run
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\thread_pool\worker.rs:318
  28: tokio::runtime::thread_pool::worker::run::{{closure}}
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\thread_pool\worker.rs:303
  29: tokio::macros::scoped_tls::ScopedKey<tokio::runtime::thread_pool::worker::Context>::set<tokio::runtime::thread_pool::worker::Context,closure-0,tuple<>>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\macros\scoped_tls.rs:61
  30: tokio::runtime::thread_pool::worker::run
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\thread_pool\worker.rs:300
  31: tokio::runtime::thread_pool::worker::{{impl}}::launch::{{closure}}
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\thread_pool\worker.rs:279
  32: tokio::runtime::blocking::task::{{impl}}::poll<closure-0,tuple<>>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\blocking\task.rs:42
  33: tokio::runtime::task::core::{{impl}}::poll::{{closure}}<tokio::runtime::blocking::task::BlockingTask<closure-0>,tokio::runtime::blocking::schedule::NoopSchedule>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\core.rs:173
  34: tokio::loom::std::unsafe_cell::UnsafeCell<tokio::runtime::task::core::Stage<tokio::runtime::blocking::task::BlockingTask<closure-0>>>::with_mut<tokio::runtime::task::core::Stage<tokio::runtime::blocking::task::BlockingTask<closure-0>>,core::task::poll::Po
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\loom\std\unsafe_cell.rs:14
  35: tokio::runtime::task::core::Core<tokio::runtime::blocking::task::BlockingTask<closure-0>, tokio::runtime::blocking::schedule::NoopSchedule>::poll<tokio::runtime::blocking::task::BlockingTask<closure-0>,tokio::runtime::blocking::schedule::NoopSchedule>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\core.rs:158
  36: tokio::runtime::task::harness::{{impl}}::poll::{{closure}}<tokio::runtime::blocking::task::BlockingTask<closure-0>,tokio::runtime::blocking::schedule::NoopSchedule>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\harness.rs:107
  37: core::ops::function::FnOnce::call_once<closure-0,tuple<>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
  38: std::panic::{{impl}}::call_once<core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>,closure-0>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:322
  39: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:379
  40: tokio::time::driver::wheel::level::slot_for
  41: std::panicking::try<core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:343
  42: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,core::task::poll::Poll<core::result::Result<tuple<>, tokio::runtime::task::error::JoinError>>>
             at C:\Users\TiTan\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:396
  43: tokio::runtime::task::harness::Harness<tokio::runtime::blocking::task::BlockingTask<closure-0>, tokio::runtime::blocking::schedule::NoopSchedule>::poll<tokio::runtime::blocking::task::BlockingTask<closure-0>,tokio::runtime::blocking::schedule::NoopSchedul
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\harness.rs:89
  44: tokio::runtime::task::raw::poll<tokio::runtime::blocking::task::BlockingTask<closure-0>,tokio::runtime::blocking::schedule::NoopSchedule>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\raw.rs:104
  45: tokio::runtime::task::raw::RawTask::poll
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\raw.rs:66
  46: tokio::runtime::task::Notified<tokio::runtime::blocking::schedule::NoopSchedule>::run<tokio::runtime::blocking::schedule::NoopSchedule>
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\task\mod.rs:171
  47: tokio::runtime::blocking::pool::Inner::run
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\blocking\pool.rs:277
  48: tokio::runtime::blocking::pool::{{impl}}::spawn_thread::{{closure}}
             at C:\Users\TiTan\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.0.1\src\runtime\blocking\pool.rs:257
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

T

**Comment by panicbit** [comment]
Minimal repro:

```rust
use thirtyfour::prelude::*;
use tokio::time::*;

#[tokio::main]
async fn main() -> WebDriverResult<()> {
    let caps = DesiredCapabilities::chrome();
    let driver = WebDriver::new("http://localhost:9515/", &caps).await?;

    let fut = driver.get("http://example.org");
    let _ = timeout(Duration::from_millis(1), fut).await;

    Ok(())
}
```

**Comment by panicbit** [comment]
The issue is that the [execute](https://github.com/stevepryde/thirtyfour/blob/dd8830dfd359ee53a3a20245761413e559170c9c/src/session.rs#L84-L103) future gets polled and then dropped before finishing. This drops the receiver and makes the [this send](https://github.com/stevepryde/thirtyfour/blob/dd8830dfd359ee53a3a20245761413e559170c9c/src/session.rs#L47) panic as a result.

**Comment by stevepryde** [comment]
Ok that makes sense. So I assume in this case it should probably just ignore the send error? Perhaps we can log it as a warning and then continue.

**Comment by panicbit** [comment]
Yes, ignoring the error is likely the way to go. We tried it for the application in the OP and it seemed to work fine.
Regarding logging, I'd probably go with a `Debug` loglevel, since this event seems fairly uninteresting outside of this crate.

**Comment by stevepryde** [comment]
Good point 👍

**Comment by Tockra** [comment]
As panicbit says after removing the expect, my application works again.

**Comment by Tockra** [comment]
@stevepryde could you notify me here, when this fix is in the crates.io version of this crate? Until then I will use a local git branch ;)

**Comment by stevepryde** [comment]
@Tockra I've uploaded a fix in `v0.23.1` which should be on crates.io now. Let me know if this fixes the issue.
